src/link_prober/LinkProber.cpp

```diff
@@ -748,4 +772,62 @@ void LinkProber::resetIcmpPacketCounts()
     )));
 }
 
+//
+// ---> reportHeartbeatReplyReceivedActiveStandby
+//
+// report heartbeat reply received to active-standby mode link prober state machine
+//
+void LinkProber::reportHeartbeatReplyReceivedActiveStandby()
+{
+    if (mTxSeqNo == mRxSelfSeqNo) {
+        mLinkProberStateMachinePtr->postLinkProberStateEvent(LinkProberStateMachineBase::getIcmpSelfEvent());
+    } else if (mTxSeqNo == mRxPeerSeqNo) {
+        mLinkProberStateMachinePtr->postLinkProberStateEvent(LinkProberStateMachineBase::getIcmpPeerEvent());
+    }
+}
+
+//
+// ---> reportHeartbeatReplyNotReceivedActiveStandby
+//
+// report heartbeat reply received to active-active mode link prober state machine
+//
+void LinkProber::reportHeartbeatReplyNotReceivedActiveStandby()
+{
+    if (mTxSeqNo != mRxSelfSeqNo && mTxSeqNo != mRxPeerSeqNo) {
+        // post unknown event
+        mLinkProberStateMachinePtr->postLinkProberStateEvent(LinkProberStateMachineBase::getIcmpUnknownEvent());
+        mIcmpUnknownEventCount++;
+    }
+}
+
+//
+// ---> reportHeartbeatReplyReceivedActiveActive
+//
+// report heartbeat reply not received to active-standby mode link prober state machine
+//
+void LinkProber::reportHeartbeatReplyReceivedActiveActive()
+{
+    if (mTxSeqNo == mRxSelfSeqNo) {
+        mLinkProberStateMachinePtr->postLinkProberStateEvent(LinkProberStateMachineBase::getIcmpSelfEvent());
+    }
+    if (mTxSeqNo == mRxPeerSeqNo) {
+        mLinkProberStateMachinePtr->postLinkProberStateEvent(LinkProberStateMachineBase::getIcmpPeerActiveEvent());
+    }
+}
+
+//
+// ---> reportHeartbeatReplyNotReceivedActiveActive
+//
+// report heartbeat reply not received to active-active mode link prober state machine
+//
+void LinkProber::reportHeartbeatReplyNotReceivedActiveActive()
+{
+    if (mTxSeqNo != mRxSelfSeqNo) {
+        mLinkProberStateMachinePtr->postLinkProberStateEvent(LinkProberStateMachineBase::getIcmpUnknownEvent());
+    }
```

Comment:
Are we planning to keep the pck loss counts for `active-active` as well? If so are we missing `mIcmpUnknownEventCount++;`  here?

Reply:
Yes, I want to leave the telemetry features to future PRs, this PR only addresses the basic event handling && state processing functionality.